src/main/java/gregtech/api/metatileentity/multiblock/MultiblockWithDisplayBase.java

```diff
@@ -255,9 +287,37 @@ public void invalidateStructure() {
                 getAbilities(MultiblockAbility.MAINTENANCE_HATCH).get(0)
                         .storeMaintenanceData(maintenance_problems, timeActive);
         }
+        this.replaceVariantBlocksActive(false);
+        this.lastActive = false;
         super.invalidateStructure();
     }
 
+    protected void replaceVariantBlocksActive(boolean isActive) {
+        if (variantActiveBlocks != null && !variantActiveBlocks.isEmpty()) {
+            int id = getWorld().provider.getDimension();
+
+            writeCustomData(GregtechDataCodes.VARIANT_RENDER_UPDATE, buf -> {
+                buf.writeInt(id);
+                buf.writeBoolean(isActive);
+                buf.writeInt(variantActiveBlocks.size());
+                for (BlockPos blockPos : variantActiveBlocks) {
+                    if (isActive) {
+                        VariantActiveBlock.ACTIVE_BLOCKS.get(id).add(blockPos);
+                    } else {
+                        VariantActiveBlock.ACTIVE_BLOCKS.get(id).remove(blockPos);
+                    }
+                    buf.writeBlockPos(blockPos);
+                    Block b = getWorld().getBlockState(blockPos).getBlock();
+                    if (b instanceof VariantActiveBlock) {
+                        if (((VariantActiveBlock<?>) b).updatesLights()) {
+                            getWorld().checkLight(blockPos);
```

Comment:
This also needs to be applied to multiblock parts. They currently stay lit after the multiblock finishes. Replication: Construct a Large Boiler, get it running, break the controller, multiblock parts are still emitting light